### PR TITLE
Support changes for "[dir]" delivery.

### PR DESCRIPTION
**Pull request by Westbrook**
## Description
- use a mutation observer of `document.documentElement` to manage `dir` on the individual components
- add `dir` support to the documentation site
- add `dir` management to the Storybook build
- expand Visual Regressions to include both `dir` values
- catch a couple of visual delivery issues in the process
  - https://github.com/adobe/spectrum-css/issues/795
  - https://github.com/adobe/spectrum-css/issues/794

## Motivation and Context
Out components should support content/functional delivery across multiple languages.

## How Has This Been Tested?
- lots and lots of visual regressions, a base line of 200 screenshots across 16 different theme combinations

## Screenshots (if appropriate):
image

## Types of changes
- [x] documentation, support, refactor

## Checklist:
- [x] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: .storybook/preview.js [hunk 1]
```diff
@@ -34,11 +34,24 @@ const scaleOptions = {
     Large: 'large',
 };
 let defaultScale = scaleOptions.Medium;
+const directionOptions = {
```
Comment (andrewhatran) [inline .storybook/preview.js:37]: This is extremely helpful 🙏 
Thanks!

**Comment by andrewhatran** [comment]
Great work! Both the storybook and documentation site look rock solid 👍 

Did we want to flip the direction of the RTL code examples for this PR? Definitely don't think it's worth blocking though.

File: packages/overlay/src/OverlayTrigger.ts [hunk 2]
```diff
@@ -179,26 +179,28 @@ export class OverlayTrigger extends LitElement {
         }
     }
 
-    private onClickSlotChange(event: Event): void {
+    private onClickSlotChange(
+        event: Event & { target: HTMLSlotElement }
```
Comment (andrewhatran) [inline packages/overlay/src/OverlayTrigger.ts:183]: I've never seen this typing before. Looks very useful

Comment (Westbrook) [reply packages/overlay/src/OverlayTrigger.ts:183]: 👍 I'm still trying to understand why _sometimes_ TS will be annoyed by this, but in places where we can this reduces boilerplate internal of the event listener that is harder to be washed out by the TS build.

**Pull request merged**